DragonSlayer.cpp

```diff
@@ -1,23 +1,39 @@
 #include "DragonSlayer.h"
 #include "Dragon.h"
+#include "Utility.h"
 
-//DragonSlayer::DragonSlayer
+DragonSlayer::DragonSlayer(std::string name_, int hp, int armor_) : Character(hp, armor_, 4),
+name(name_)
+{
+    helpfulItems = makeHelpfulItems(3);
+    defensiveItems = makeDefensiveItems(1);
+}
 
-//DragonSlayer::getName
+const std::string& DragonSlayer::getName()
+{
+    return name;
+}
 
 void DragonSlayer::attack(Character& other)
 {
     std::cout << name << " is attacking " << other.getName() << " !!" << std::endl;
     if( auto* dragon = dynamic_cast<Dragon*>(&other) )
-    {
-        assert(false);
-        //DragonSlayers get a 10x boost when attacking dragons, from their attack item.
-        //so they should USE their attack item before attacking the dragon... 
-        //note: they should only use the item if the dragon's hitpoints are > 0...
-        //note: items are single-use only, so you need to reset it after use.  
-        //look in the Character class for how the other item types are reset after use.
+    {   
         while( dragon->getHP() > 0 )
         {
+            //DragonSlayers get a 10x boost when attacking dragons, from their attack item.
+
+            if (attackItem != nullptr)
+            {
+                //so they should USE their attack item before attacking the dragon... 
+                attackItem->use(this);
+                //note: items are single-use only, so you need to reset it after use.  
+                //look in the Character class for how the other item types are reset after use.
+                attackItem.reset();
+            }
```

Comment:
Use the attack item before the while() loop.
if you don't, you'll be performing this check every time this loop runs, which is redundant after the first usage.
- [ ] revise

Reply:
I've made the change